components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -611,7 +611,7 @@ private void validateBindingType(String bindingType) throws IdentityOAuthClientE
     private void validateFAPIBindingType(String bindingType)
             throws IdentityOAuthClientException {
 
-        if (OAuth2Constants.TokenBinderType.CERTIFICATE_BASED_TOKEN_BINDER.equals(bindingType)) {
+        if (OAuth2Constants.TokenBinderType.CERTIFICATE_BASED_TOKEN_BINDER.equals(bindingType) || bindingType == null) {
```

Comment:
By default FAPI app is should bound to CERTIFICATE_BASED_TOKEN_BINDER type, why we are allowing null type now?

Reply:
FAPI apps created from UI are created without the initial configurations and once the user updates the protocol configs page accordingly the app will be FAPI compliant. From UI side we have already checked this value and is unchangeable. 
Having null initially will allow to create the FAPI enabled app at first place bypassing the validation.